xtask/src/themelint.rs

```diff
@@ -0,0 +1,172 @@
+        Rule::has_either("ui.window".into()),
+        // Check for visible cursor
+        Rule::has_bg("ui.cursor.primary".into()),
+        Rule::has_either("ui.cursor.match".into()),
+    ]
+}
+
+impl Rule {
+    fn has_fg_bg(fg: String, bg: String) -> Rule {
+        Rule {
+            fg: Some(fg),
+            bg: Some(bg),
+            check_both: true,
+        }
+    }
+    fn has_bg(bg: String) -> Rule {
+        Rule {
+            fg: None,
+            bg: Some(bg),
+            check_both: true,
+        }
+    }
+    fn has_fg(fg: String) -> Rule {
+        Rule {
+            fg: Some(fg),
+            bg: None,
+            check_both: true,
+        }
+    }
+    fn has_either(item: String) -> Rule {
+        Rule {
+            fg: Some(item.clone()),
+            bg: Some(item),
+            check_both: false,
+        }
+    }
+    fn has_both(item: String) -> Rule {
+        Rule {
+            fg: Some(item.clone()),
+            bg: Some(item),
+            check_both: true,
+        }
+    }
+    fn validate(&self, theme: &Theme, messages: &mut Vec<String>) {
+        let mut found_fg = true;
+        let mut found_bg = true;
+        let mut fg_name = "";
+        let mut bg_name = "";
+        if let Some(fg) = &self.fg {
+            fg_name = fg;
+            if theme.get(fg).fg.is_none() && theme.get(fg).add_modifier == Modifier::empty() {
+                found_fg = false;
+            }
+        }
+        if let Some(bg) = &self.bg {
+            bg_name = bg;
+            if theme.get(bg).bg.is_none() && theme.get(bg).add_modifier == Modifier::empty() {
+                found_bg = false;
+            }
+        }
+        if self.check_both {
+            if !found_fg {
+                messages.push(format!("{}.fg", fg_name.clone()));
+            }
+            if !found_bg {
+                messages.push(format!("{}.bg", bg_name.clone()));
+            }
+        } else {
+            if !found_fg && !found_bg {
+                messages.push(format!("{}", fg_name))
+            }
+        }
+    }
+}
+
+pub fn lint(file: String) -> Result<(), DynError> {
+    if file.contains("base16") {
+        println!("Skipping base16: {}", file);
+        return Ok(());
+    }
+    let path = path::themes().join(file.clone() + ".toml");
+    let theme = std::fs::read(&path).unwrap();
+    let theme: Theme = toml::from_slice(&theme).expect("Failed to parse theme");
+
+    let mut messages: Vec<String> = vec![];
+    get_rules()
+        .iter()
+        .for_each(|rule| rule.validate(&theme, &mut messages));
+
+    if messages.len() > 0 {
+        let message: String = messages
+            .iter()
+            .map(|m| {
+                let mut msg = file.clone();
+                msg.push_str(".");
+                msg.push_str(m);
+                msg
+            })
+            .collect::<Vec<String>>()
+            .join(" ")
+            .into();
+        println!("{}", message.replace(" ", "\n"));
```

Comment:
If I am a new contributor to theme, just seeing
```
everforest_light.ui.cursorline.primary.bg
everforest_light.ui.virtual.ruler
```
I have no idea what I did wrong there. At least some explanation on why is that error there is better, a better version would be like rust sort of error where it point you out the line and span, but showing at least the error is a good start.

Reply:
```sh
gruvbox:"ui.cursorline.primary"={"bg"="missing"}
gruvbox:"ui.virtual.ruler"="missing"
```
This is better, yes? Now you get copy pastable "code" after the theme key, which says missing.
If you've got ideas to improve it further (maybe adding a text to each rule with an actual feedback?) I'm happy to implement!

Reply:
Better slightly but still can be improved, I imagine something like
```
Missing `bg` for gruvbox `ui.cursorline.primary`
```
Also, not sure why we want to split `fg` and `bg` reporting when they can be in the same line, like `Missing bg and fg for ...`.

Reply:
Marking this as resolved for now```shell
gruvbox: missing `fg` and `bg` for `ui.statusline`
gruvbox: missing entry for `ui.virtual.ruler`
```
Marking this s resolved for now.